core/db/migrate/20130830001159_migrate_old_shipping_calculators.rb

```diff
@@ -1,15 +1,28 @@
 class MigrateOldShippingCalculators < ActiveRecord::Migration
   def up
     Spree::ShippingMethod.all.each do |shipping_method|
-      old_calculator = shipping_method.calculator
-      next if old_calculator.class < Spree::ShippingCalculator # We don't want to mess with new shipping calculators
-      new_calculator = eval(old_calculator.class.name.sub("::Calculator::", "::Calculator::Shipping::")).new
+      old_calculator_id = ActiveRecord::Base.connection.select_values(
+        "SELECT id FROM spree_calculators WHERE calculable_type = 'Spree::ShippingMethod' AND calculable_id = #{shipping_method.id}"
+      ).first
+      old_calculator_class_name = ActiveRecord::Base.connection.select_values(
+        "SELECT type FROM spree_calculators WHERE id = '#{old_calculator_id}'"
+      ).first
+      next if old_calculator_class_name.start_with? 'Spree::Calculator::Shipping'
+      old_calculator_calculable_id = ActiveRecord::Base.connection.select_values(
+        "SELECT calculable_id FROM spree_calculators WHERE id = '#{old_calculator_id}'"
+      ).first
+
+      new_calculator = eval(old_calculator_class_name.sub("::Calculator::", "::Calculator::Shipping::")).new
+      new_calculator.calculable_id = old_calculator_calculable_id
+      new_calculator.calculable_type = 'Spree::ShippingMethod'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.